### PR TITLE
fixing unintended behaviour for #11455

### PR DESCRIPTION
**Pull request by OfekShochat**

### CODE REVIEW

**Comment by luehmann** [comment]
Hi @OfekShochat, I think it would make sense to add tests for this. You can take a look the pull-request I opened before I saw yours.

**Comment by OfekShochat** [comment]
Ok will do! Busy right now though, might take some time

File: lib/std/json.zig [hunk 1]
```diff
@@ -3048,7 +3048,9 @@ pub fn stringify(
                         try out_stream.writeByte('\n');
                         try child_whitespace.outputIndent(out_stream);
                     }
-                    try stringify(Field.name, options, out_stream);
+                    try out_stream.writeByte('\"');
+                    try out_stream.writeAll(Field.name);
```
Comment (daurnimator) [inline lib/std/json.zig:3052]: This is not correct; it may need escaping.

**Review by daurnimator — CHANGES_REQUESTED** [review]
Make sure you add a test case of a key that needs escaping in json.

**Comment by andrewrk** [comment]
* CI tests failing
 * Conflicts with master branch
 * No activity for 28 days

Please open a fresh PR if you would like to continue this effort.

**Pull request closed without merging**